### PR TITLE
Icon refactor

### PR DESCRIPTION
**Pull request by binarysudoku**
## About the PR

Same PR as last time but now to the right branch, and also with the sprite changes ZeWaka requested

### CODE REVIEW

**Comment by binarysudoku** [comment]
PLEASE IGNORE THIS WHILE I SOLVE MERGE CONFLICTS

**Comment by binarysudoku** [comment]
OK
I THINK
THIS IS OK NOW

**Comment by binarysudoku** [comment]
IT WORKS
I'M THE SMARTEST PERSON IN THE ENTIRE WORLD
THE SCOPE OF MY ENGINEERING GENIUS KNOWS LITERALLY NO BOUNDS

File: code/modules/food_and_drink/discount_dans.dm.orig [hunk 1]
```diff
@@ -0,0 +1,594 @@
+
```
Comment (ZeWaka) [inline code/modules/food_and_drink/discount_dans.dm.orig:1]: errant file, please delete

File: code/obj/item/kitchen.dm.orig [hunk 2]
```diff
@@ -0,0 +1,1182 @@
+/*
```
Comment (ZeWaka) [inline code/obj/item/kitchen.dm.orig:1]: errant file, please delete

File: code/modules/food_and_drink/discount_dans.dm.orig (hunk 1 above)
Comment (binarysudoku) [reply code/modules/food_and_drink/discount_dans.dm.orig]: Deleted errant file in latest CL (ca59a02)

File: code/obj/item/kitchen.dm.orig (hunk 2 above)
Comment (binarysudoku) [reply code/obj/item/kitchen.dm.orig]: Deleted in most recent CL (744a409)

File: code/obj/item/kitchen.dm [hunk 3]
```diff
@@ -887,14 +887,11 @@ TRAYS
 						var/image/rolltopping = new /image('icons/obj/kitchen.dmi',"roll_topping-[i]")
 						switch(i)
 							if(1)
-								if(topping1)
-									rolltopping.color = topping1.food_color
+								rolltopping.color = topping1.food_color
```
Comment (pali6) [inline code/obj/item/kitchen.dm:890]: This is a regression, please fix.

Comment (binarysudoku) [reply code/obj/item/kitchen.dm]: Undid the regression in commit 69cf469

File: code/modules/food_and_drink/plants.dm [hunk 4]
```diff
@@ -509,8 +509,8 @@
 	crop_prefix = "bowling "
 	desc = "Just keep rollin' rollin'."
 	icon = 'icons/obj/items.dmi'
```
Comment (pali6) [inline code/modules/food_and_drink/plants.dm:511]: It seems that this should be `icons/obj/foodNdrink/food_produce.dmi` now.

Comment (binarysudoku) [reply code/modules/food_and_drink/plants.dm]: Fixed!

**Pull request merged**